lib/rss.js

```diff
@@ -0,0 +1,192 @@
+const RSS = require('rss')
+const generateDiff = require('deep-diff').diff
+const fs = require('fs')
+
+function createLeadersHash(array) {
+  return array.reduce((hash, elem) => {
+    hash[elem.id] = elem
+    return hash
+  }, {})
+}
+
+function createOrgHash(array) {
+  let rank = 0
+  return array.reduce((hash, elem) => {
+    rank++
+    elem.rank = rank
+    elem.leaders = createLeadersHash(elem.leaders)
+    hash[elem.slug] = elem
+    return hash
+  }, {})
+}
+
+let current = {}
+try {
+  current = JSON.parse(
+    fs.readFileSync(`${__dirname}/../out/feed_items.json`).toString()
+  )
+} catch (e) {
+  current = {}
+}
+
+const buildInfo = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/scrape_info.json`).toString()
+)
+
+const oldData = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/data_old.json`).toString()
+)
+const newData = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/data.json`).toString()
+)
+
+const oldOrgs = createOrgHash(oldData)
+const newOrgs = createOrgHash(newData)
+
+console.log(oldOrgs)
+
+if (oldOrgs && newOrgs) {
+  const diffs = generateDiff(oldOrgs, newOrgs) || []
+  console.log(diffs)
+
+  const feedItems = current.items || []
+
+  if (current.lastUpdated !== buildInfo.generated) {
```

Comment:
how can they ever be equal!?

more importantly, you do not need to store the date generated.  The `out/` files have dates, and HTTP tells you the dates of those files.

Reply:
>HTTP tells you the dates of those files.

It does, but not all host will serve the correct date, e.g. netlify.

Reply:
But yeah, it'll never be equal.
So no need to read the date.

Reply:
The value of `current.lastUpdated` is set to `buildInfo.generated` [here](https://github.com/coala/gci-leaders/pull/95/files#diff-9676cbcb5938ab810f7fa82a528abc4eR183) when the file is created, so they will be equal if the same `data.json` file is being run multiple times.

Reply:
netlify should serve the correct date.  We need to set the date on the files so they stay constant even after regeneration , if the content is the same.